packages/block-library/src/navigation/edit/use-fallback.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { useDispatch } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import { addQueryArgs } from '@wordpress/url';
+import { useRef } from '@wordpress/element';
+
+export default function useFallback() {
+	const isFetching = useRef( false );
+	const { receiveEntityRecords } = useDispatch( coreStore );
+
+	return async function () {
+		if ( isFetching?.current ) {
+			return;
+		}
+
+		isFetching.current = true;
+
+		const fallback = await apiFetch( {
+			path: addQueryArgs( '/wp-block-editor/v1/navigation-fallback', {
+				_embed: true,
+			} ),
+		} );
+
+		const record = fallback?._embedded?.self;
+
+		if ( record ) {
```

Comment:
I added a `debugger` here _after_ the `receiveEntityRecords ` call and then switched to `console` and ran:

```
wp.data.select('core').getEntityRecord('postType', 'wp_navigation', %%THE_ID%%)
```

...and the record is in state.

So why - with this non-Core data approach - do we get a 2nd network request coming from:

https://github.com/WordPress/gutenberg/blob/fedc841f3d77c430455e0054cca26a9f61a1ab70/packages/block-library/src/navigation/use-navigation-menu.js#L98

Reply:
Consider calling

```
dispatch.finishResolution( 'getEntityRecord', [
				'postType',
				'wp_navigation',
				fallback?.id,
			] );
```